### PR TITLE
Add module example app

### PR DESCRIPTION
**Pull request by marekrozmus**

### CODE REVIEW

File: examples/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "name": "simple",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "build": "webpack --mode production",
+    "start": "webpack-dev-server"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (przemyslawzalewski) [inline examples/package.json]: MIT?
```suggestion
  "license": "MIT",
```

File: examples/src/App.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
+import React, { useState } from 'react';
+import { SwipeableList, SwipeableListItem } from 'react-swipeable-list';
+
+import ListItem from './ListItem';
+import ComplexItemBackground from './ComplexItemBackground';
+import { ReactComponent as MailIcon } from '../images/mail.svg';
```
Comment (przemyslawzalewski) [inline examples/src/App.js]: Please investigate if the weird named export can be fixed in favor of regular default import.

File: examples/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "env": {
+    "browser": true,
+    "node": true
```
Comment (przemyslawzalewski) [inline examples/.eslintrc]: Just browser there as this is a purely client-side example app.

```suggestion
```

File: examples/.eslintrc [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  },
```
Comment (przemyslawzalewski) [inline examples/.eslintrc]: I think the example should be minimal - no prettier, no eslint, just raw usage with plain html/webpack at most.

File: examples/src/App.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
+import React, { useState } from 'react';
+import { SwipeableList, SwipeableListItem } from 'react-swipeable-list';
+
+import ListItem from './ListItem';
+import ComplexItemBackground from './ComplexItemBackground';
```
Comment (przemyslawzalewski) [inline examples/src/App.js]: Please sort by relativity if possible.

File: examples/src/App.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+import './app.css';
+
+function App() {
+  const [triggeredSimpleItemAction, triggerSimpleItemAction] = useState('');
+  const [triggeredComplexItemAction, triggerComplexItemAction] = useState('');
+
+  return (
+    <div className="example">
+      <h3>react-swipeable-list example</h3>
+      <h5>(switch on dev tools to mobile view)</h5>
```
Comment (przemyslawzalewski) [inline examples/src/App.js]: Can we use some kind of wrapper to toggle between views on the page itself?

File: examples/src/ComplexItemBackground.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './ComplexItemBackground.css';
+
+const ComplexItemBackground = ({ icon, color, side, label }) => (
+  <div className={`complex-list-item-background ${color} ${side}`}>
+    <div className="content">
+      {icon && icon}
```
Comment (przemyslawzalewski) [inline examples/src/ComplexItemBackground.js]: Just:

```suggestion
      {icon}
```

File: examples/src/ComplexItemBackground.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      {label && <span>{label}</span>}
+    </div>
+  </div>
+);
+
+ComplexItemBackground.propTypes = {
+  color: PropTypes.string.isRequired,
+  icon: PropTypes.node,
+  label: PropTypes.string,
+  side: PropTypes.string.isRequired
```
Comment (przemyslawzalewski) [inline examples/src/ComplexItemBackground.js]: This should be enum if there are only a few possible options.

File: examples/src/ListItem.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import './ListItem.css';
+
+class ListItem extends PureComponent {
```
Comment (przemyslawzalewski) [inline examples/src/ListItem.js]: Plain function component would be good fit for this.

File: examples/src/index.js [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
```
Comment (przemyslawzalewski) [inline examples/src/index.js]: Please split external and internal import groups.

File: examples/webpack.config.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+var path = require('path');
+
+const babelLoaderOptions = {
+  presets: ['@babel/env', '@babel/react'],
+  plugins: ['@babel/plugin-proposal-class-properties']
```
Comment (przemyslawzalewski) [inline examples/webpack.config.js]: Is this used?

File: examples/package.json [hunk 12]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "simple",
```
Comment (przemyslawzalewski) [inline examples/package.json]: SwipeableList

File: examples/package.json [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+    "style-loader": "^1.0.0",
+    "stylelint": "^10.1.0",
+    "stylelint-config-standard": "^18.3.0",
+    "url-loader": "^2.1.0",
+    "webpack": "^4.39.3",
+    "webpack-cli": "^3.3.8",
+    "webpack-dev-server": "^3.8.0"
+  },
+  "dependencies": {
+    "react": "^16.9.0",
```
Comment (jakubbogacz) [inline examples/package.json]: shouldn't we add ```react-swipeable-list``` as a dependency?

File: examples/src/App.js (hunk 6 above)
Comment (jakubbogacz) [reply examples/src/App.js]: Agree. Is it possible to prepare the sample to work on the demo page, without the need to open dev-tools and switching to mobile mode?

File: examples/src/index.html [hunk 14]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <link
+      href="https://fonts.googleapis.com/css?family=Roboto&display=swap"
+      rel="stylesheet"
+    />
+    <meta
```
Comment (jakubbogacz) [inline examples/src/index.html]: Maybe we could add some additional SEO meta tags?

File: examples/package.json (hunk 13 above)
Comment (marekrozmus) [reply examples/package.json]: no, it just uses sources, not a module

**Pull request merged**